examples/rcn.py

```diff
@@ -0,0 +1,392 @@
+# # 2. Load the model
+
+# %%
+data = np.load(
+    f"example_data/rcn_{train_size}.npz", allow_pickle=True, encoding="latin1"
+)
+frcs, edges, suppression_masks, filters = (
+    data["frcs"][:train_size],
+    data["edges"][:train_size],
+    data["suppression_masks"],
+    data["filters"],
+)
+
+M = (2 * hps + 1) * (2 * vps + 1)
+
+# %% [markdown]
+# # 3. Visualize loaded model
+
+# %%
+img = np.ones((200, 200))
+pad = 55
+frc, edge = frcs[4], edges[4]
+fig, ax = plt.subplots(1, 1, figsize=(10, 10))
+for e in edge:
+    i1, i2, w = e
+    f1, r1, c1 = frc[i1]
+    f2, r2, c2 = frc[i2]
+
+    img[r1, c1] = 0
+    img[r2, c2] = 0
+    ax.text(
+        (c1 + c2) // 2 - pad, (r1 + r2) // 2 - pad, str(w), color="green", fontsize=15
+    )
+    ax.plot([c1 - pad, c2 - pad], [r1 - pad, r2 - pad], color="green", linewidth=0.5)
+
+ax.axis("off")
+ax.imshow(img[pad : 200 - pad, pad : 200 - pad], cmap="gray")
+
+
+# %% [markdown]
+# ## 3.1 Visualize the filters
+
+# %% [markdown]
+# The filters are used to detect the oriented edges on a given image. They are pre-computed using Gabor filters.
+
+# %%
+fig, ax = plt.subplots(4, 4, figsize=(10, 10))
+for i in range(filters.shape[0]):
+    idx = np.unravel_index(i, (4, 4))
+    ax[idx].imshow(filters[i], cmap="gray")
+    ax[idx].axis("off")
+
+fig.tight_layout()
+
+# %% [markdown]
+# # 4. Make pgmax graph
+
+# %% [markdown]
+# ## 4.1 Make variables
+
+# %%
+start = time.time()
+assert frcs.shape[0] == edges.shape[0]
+
+variables_all_models = {}
+for idx in range(frcs.shape[0]):
+    frc = frcs[idx]
+    variables_all_models[idx] = groups.NDVariableArray(
+        variable_size=M, shape=(frc.shape[0],)
+    )
+
+end = time.time()
+print(f"Creating variables took {end-start:.3f} seconds.")
+
+
+# %% [markdown]
+# ## 4.2 Make factors
+
+# %% [markdown]
+# ### 4.2.1 Pre-compute the valid configs for different perturb radii.
+
+# %%
+def valid_configs(r: int) -> np.ndarray:
+    """Returns the valid configurations for the potential matrix given the perturb radius.
+
+    Args:
+        r: Peturb radius
+
+    Returns:
+        A configuration matrix (shape n X 2) where n is the number of valid configurations.
+    """
+
+    configs = []
+    for i, (r1, c1) in enumerate(np.array(np.unravel_index(np.arange(M), (25, 25))).T):
+        r2_min = max(r1 - r, 0)
+        r2_max = min(r1 + r, 2 * hps)
+        c2_min = max(c1 - r, 0)
+        c2_max = min(c1 + r, 2 * vps)
+        j = np.ravel_multi_index(
+            tuple(np.mgrid[r2_min : r2_max + 1, c2_min : c2_max + 1]), (25, 25)
+        ).ravel()
+        configs.append(np.stack([np.full(j.shape, fill_value=i), j], axis=1))
+
+    return np.concatenate(configs)
+
+
+max_perturb_radius = 25
+valid_configs_list = [valid_configs(r) for r in range(max_perturb_radius)]
+
+# %% [markdown]
+# ### 4.2.2 Make the factor graph
+
+# %%
+start = end
+fg = graph.FactorGraph(variables=variables_all_models)
+for idx in range(edges.shape[0]):
+    edge = edges[idx]
+
```

Comment:
Worth adding a comment on what you are adding here like how you are mapping RCN model edges to factor graph.

Reply:
Added comment.